addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -379,10 +381,9 @@ export const editorCommands = {
                 children.forEach(child => newBlock.appendChild(child));
             }
         }
-        const newRange = new Range();
-        newRange.setStart(startContainer,startOffset);
-        newRange.setEnd(endContainer,endOffset);
-        getDeepRange(editor.editable, { range: newRange, select: true, });
+        startContainer = !startContainer.isConnected || ['TD', 'DIV', 'LI'].includes(startContainer.nodeName) ? startContainerChild.parentNode : startContainer;
+        endContainer = !endContainer.isConnected || ['TD', 'DIV', 'LI'].includes(endContainer.nodeName) ? endContainerChild.parentNode : endContainer;
+        setSelection(startContainer, startOffset, endContainer, endOffset);
```

Comment:
When converting from one block tag to another after shift+enter for eg: `<p>abc<br>[]</p>` here the startContainer would be the `p` tag so converting it to `setTag('h1')` the `p` tag would no longer be present it dom.

Reply:
Since we call `setTag` just above (in the `for` loop) and that's where we lose the reference, it could be helped by updating the reference there. But if we're sure there is no risk of disconnecting the container children (I checked and if I'm not mistaken it's currently the case) I guess it's ok. This is clearer albeit potentially less robust, but we'll get an error if it's not connected so if a change ever makes disconnecting the container children possible, we should be warned.

tldr: ok with this change.